FWCore/ServiceRegistry/interface/ESParentContext.h

```diff
@@ -0,0 +1,43 @@
+#ifndef FWCore_ServiceRegistry_ESParentContext_h
+#define FWCore_ServiceRegistry_ESParentContext_h
+
+/**\class edm::ESParentContext
+
+ Description: This is intended to be used as a member of ESModuleCallingContext.
+
+ Usage:
+
+
+*/
+//
+// Original Author: C. Jones
+//         Created: 2/07/2021
+
+namespace edm {
+
+  class ModuleCallingContext;
+  class ESModuleCallingContext;
+
+  class ESParentContext {
```

Comment:
what about using an `std::variant<ModuleCallingContext const*, ESModuleCallingContext const*>` ?

Reply:
I've used a dedicated class since
- it follows the pattern of all the other Context classes
- dedicated types are very handy when doing debugging and having to see the actual type signature
- dedicated member function names are easier to understand in code than having to specify the type or an index value
- a union of pointers is pretty trivial to handle and doesn't need all the mechanisms available in `std::variant`

Reply:
I see. I guess the comments I receved from Core Software, that could be paraphrased here as

> We should still encourage [code] that is modern, encouraged by the standard, and has advantages over [C unions]

do not apply to Core Software developers.